### PR TITLE
upgrade otel version to 1.24

### PR DESCRIPTION
**Pull request by shashank11p**
## Description
Please include a summary of the change, motivation and context.

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

File: instrumentation/build.gradle.kts [hunk 1]
```diff
@@ -26,8 +26,15 @@ tasks {
...
         dependencies{
-            // exclude core, it lives in the bootstrap classloader
+            // exclude packages that live in the bootstrap classloader
             exclude(project(":javaagent-core"))
+            exclude(project(":filter-api"))
+            exclude("io/opentelemetry/semconv/**")
+            exclude("io/opentelemetry/context/**")
+            exclude(dependency("io.opentelemetry:opentelemetry-api"))
+            exclude("io/opentelemetry/instrumentation/api/**")
+//            // exclude bootstrap part of javaagent-extension-api
```
Comment (tim-mwangi) [inline instrumentation/build.gradle.kts:36]: nit: double comment

File: testing-bootstrap/build.gradle.kts [hunk 2]
```diff
@@ -12,12 +12,12 @@ val versions: Map<String, String> by extra
 dependencies {
     api("io.opentelemetry.javaagent:opentelemetry-javaagent-bootstrap:${versions["opentelemetry_java_agent"]}")
     api("io.opentelemetry.javaagent:opentelemetry-javaagent-extension-api:${versions["opentelemetry_java_agent"]}")
-    api("io.opentelemetry.javaagent:opentelemetry-javaagent-instrumentation-api:${versions["opentelemetry_java_agent"]}")
-    api("io.opentelemetry.instrumentation:opentelemetry-instrumentation-appender-api-internal:${versions["opentelemetry_java_agent"]}")
+    api("io.opentelemetry.instrumentation:opentelemetry-instrumentation-api:${versions["opentelemetry"]}")
+//    api("io.opentelemetry.instrumentation:opentelemetry-instrumentation-appender-api-internal:${versions["opentelemetry"]}")
```
Comment (tim-mwangi) [inline testing-bootstrap/build.gradle.kts:16]: remove?

File: instrumentation/jaxrs-client-2.0/src/test/java/io/opentelemetry/javaagent/instrumentation/hypertrace/jaxrs/v2_0/JaxrsAsyncClientBodyInstrumentationTest.java [hunk 3]
```diff
@@ -15,98 +15,99 @@
...
-
-    @Override
-    public boolean isWriteable(
-        Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType) {
-      return true;
-    }
-  }
-}
+//
+// import java.io.IOException;
```
Comment (tim-mwangi) [inline instrumentation/jaxrs-client-2.0/src/test/java/io/opentelemetry/javaagent/instrumentation/hypertrace/jaxrs/v2_0/JaxrsAsyncClientBodyInstrumentationTest.java:19]: Guessing this is wip?

File: instrumentation/jaxrs-client-2.0/src/test/java/io/opentelemetry/javaagent/instrumentation/hypertrace/jaxrs/v2_0/JaxrsClientBodyInstrumentationTest.java [hunk 4]
```diff
@@ -15,53 +15,54 @@
...
-    javax.ws.rs.core.Response response = builder.get();
-
-    String responseBody = response.readEntity(String.class);
-
-    return new Response(
-        responseBody == null || responseBody.isEmpty() ? null : responseBody, response.getStatus());
-  }
-}
+//
+// import java.util.Map;
```
Comment (tim-mwangi) [inline instrumentation/jaxrs-client-2.0/src/test/java/io/opentelemetry/javaagent/instrumentation/hypertrace/jaxrs/v2_0/JaxrsClientBodyInstrumentationTest.java:19]: wip?

**Comment by tyroneyu** [comment]
Aside from the commented code Tim pointed out, lgtm.

**Comment by shashank11p** [comment]
Removed jaxrs instrumentation since it did not provide any additional value to us. The client spans are already being created by apache-httpclient instrumentation. And the passing tests show that we will be able to capture all that we need without this extra instrumentation as well

**Pull request merged**